### PR TITLE
Make api requests return chaffed json data.

### LINKED ISSUE #1155: Return JSON response bodies on Chaff requests
### TL;DR

The go-chaff library already supports this
  https://github.com/mikehelmick/go-chaff/blob/main/json.go

Use this to respond w/ properly formed JSON based on the request being chaffed - for our APIs, this really just means having a `padding` key in the JSON object.

REMEMBER - the size of the random data will need to be reduced as base 64 encoding will cause ~33% size increase.

/priority important-soon
/kind enhancement

### PR DESCRIPTION
**Pull request by jeremyfaller**
Fixes #1155

```release-note
Chaff v1 api requests now return chaffed JSON data.
```

### CODE REVIEW

File: internal/publish/publish.go [hunk 1]
```diff
@@ -444,3 +449,9 @@ func (h *PublishHandler) process(ctx context.Context, data *verifyapi.Publish, b
 		},
 	}
 }
+
+// chaffPushResponse takes a chaffing string, and builds a chaff response.
+func chaffPublishResponse(s string) interface{} {
+	return verifyapi.PublishResponse{Padding: s[:(len(s)*7)/10]}
```
Comment (mikehelmick) [inline internal/publish/publish.go:455]: the chaff library already accounts for base64 encoding

Comment (mikehelmick) [reply internal/publish/publish.go:455]: and doing a sub-slice here might make this invalid base64

**Review by mikehelmick — CHANGES_REQUESTED** [review]

File: internal/publish/publish.go (hunk 1 above)
Comment (jeremyfaller) [reply internal/publish/publish.go]: I didn't realize anyone was ever going to decode the chaff. In any case, removed.

File: internal/publish/publish.go [hunk 2]
```diff
@@ -444,3 +449,9 @@ func (h *PublishHandler) process(ctx context.Context, data *verifyapi.Publish, b
 		},
 	}
 }
+
+// chaffPushResponse takes a chaffing string, and builds a chaff response.
+func chaffPublishResponse(s string) interface{} {
+	return verifyapi.PublishResponse{Padding: s}
+
```
Comment (mikehelmick) [inline internal/publish/publish.go:456]: nit: extra blank line

Comment (jeremyfaller) [reply internal/publish/publish.go]: done.

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**